File: backend/app/voice/schemas/voice.py
"""Pydantic schemas for the Voice API endpoints."""

from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field


class AudioRecordResponse(BaseModel):
    id: int
    uuid: UUID
    owner_uuid: UUID
    original_filename: str
    mime_type: str
    file_size: int
    duration_seconds: float
    language_detected: str | None = None
    language_confidence: float | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class TranscriptResponse(BaseModel):
    id: int
    uuid: UUID
    audio_uuid: UUID
    raw_transcript: str
    detected_language: str
    language_confidence: float
    transcription_confidence: float
    model_name: str
    model_version: str
    normalized_query: str | None = None
    detected_intent: str | None = None
    extracted_entities: list[dict] | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class VoiceQueryResponse(BaseModel):
    """Structured response returned by POST /voice/query."""
    request_id: UUID
    audio_id: int
    audio_uuid: UUID
    detected_language: str
    raw_transcript: str
    normalized_query: str
    response_text: str
    spoken_audio_reference: UUID | None = None
    citations: list[dict] = Field(default_factory=list)
    confidence: float
    is_code_switched: bool = False
    processing_time_ms: float = 0.0
    agent_used: str = "crop_advisory_agent"

    model_config = {"from_attributes": True}

File: frontend/src/types/voice.ts
/**
 * TypeScript types for Multilingual Voice Intelligence matching backend schemas.
 *
 * See: backend/app/voice/schemas/voice.py
 */

export interface AudioRecordResponse {
  id: number;
  uuid: string;
  owner_uuid: string;
  original_filename: string;
  mime_type: string;
  file_size: number;
  duration_seconds: number;
  language_detected?: string | null;
  language_confidence?: number | null;
  created_at: string;
}

export interface TranscriptResponse {
  id: number;
  uuid: string;
  audio_uuid: string;
  raw_transcript: string;
  detected_language: string;
  language_confidence: number;
  transcription_confidence: number;
  model_name: string;
  model_version: string;
  normalized_query?: string | null;
  detected_intent?: string | null;
  extracted_entities?: Record<string, unknown>[] | null;
  created_at: string;
}

export interface VoiceQueryResponse {
  request_id: string;
  audio_id: number;
  audio_uuid: string;
  detected_language: string;
  raw_transcript: string;
  normalized_query: string;
  response_text: string;
  spoken_audio_reference?: string | null;
  citations: Array<{
    source_title?: string;
    authority?: string;
    document_type?: string;
    snippet?: string;
    [key: string]: unknown;
  }>;
  confidence: number;
  is_code_switched: boolean;
  processing_time_ms: number;
  agent_used: string;
}
